### PR TITLE
Success page & stripe/booking logic

### PR DESCRIPTION
**Pull request by adambrodin**
- Rewrote & implemented functionality for booking a trip, paying with Stripe and then fetching checkout/receipt data in the frontend
- Added a backend endpoint for fetching booking metadata via Stripe
- Fixed issues when booking with both a outbound-trip and a return-trip
- Refactoring

### **This pull-request does not include any graphical changes to the success page, the data is only fetched, not displayed!**

### CODE REVIEW

**Comment by mzarabi** [comment]
Great work and nice code refactoring, good job!

File: client/src/PaymentPage/Payment.jsx [hunk 1]
```diff
@@ -115,156 +104,222 @@ function Payment() {
...
       .catch((e) => {
-        console.log(e.error);
+        console.log("Fetch checkout-session error:", e.error);
       });
   }
 
   function returnPage() {
-    if (context.SelectedTrain.class === 'FirstClass') {
-      nav('/results');
+    if (context.SelectedTrain.class === "FirstClass") {
```
Comment (mzarabi) [inline client/src/PaymentPage/Payment.jsx]: You forgot to change it to "First Class" here, it goes back to additionalpage even if i've selected firstclass tickets.

Comment (adambrodin) [reply client/src/PaymentPage/Payment.jsx]: Good catch, issue has been fixed.

**Review by Kristianjimmefors — APPROVED** [review]
Allt ser bra ut

**Pull request merged**